bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/api/graphql/HydrawiseGraphQLClient.java

```diff
@@ -107,15 +108,18 @@ public HydrawiseGraphQLClient(HttpClient httpClient, OAuthClientService oAuthSer
      */
     public @Nullable QueryResponse queryControllers()
             throws HydrawiseConnectionException, HydrawiseAuthenticationException {
-        QueryRequest query;
         try {
-            query = new QueryRequest(getQueryString());
+            QueryRequest query = new QueryRequest(getQueryString());
+            String queryJson = gson.toJson(query);
+            String response = sendGraphQLQuery(queryJson);
+            try {
+                return gson.fromJson(response, QueryResponse.class);
+            } catch (JsonSyntaxException e) {
```

Comment:
Not trusting the returned content by catching `JsonSyntaxException` here is wise, but wouldn't it it be even better if `sendGraphQLRequest` would also throw an exception if the HTTP request failed?

https://github.com/openhab/openhab-addons/blob/3d7fa51c65534c3c759c0f526b820f032937f29b/bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/api/graphql/HydrawiseGraphQLClient.java#L289-L304

Something like this could be added just before the last quoted line:

```java
            int status = response.getStatus();
            if (!HttpStatus.isSuccess(status)) {
                throw new HydrawiseConnectionException("Request failed with HTTP status code " + status);
            }
```

Reply:
I overlooked `onResponseFailure`. I'm not familiar with that, but if this is called on HTTP failure, my comment would need to be revised.

Reply:
yeah, there's code to catch non 200 responses (well maybe just 40x errors now that i look at it again)  , its a little tricky as the error response is not formed to http standards, and jetty throws an execution exception (commented in the code), so we throw there, at least when dealing with 401/402 errors, but now I'm not sure that other errors are being caught in that same exception block, so i'll add some additional checking code  as a precaution.

Reply:
That sounds familiar, probably same issue as I encountered here?
https://github.com/openhab/openhab-addons/blob/3f6b2b72319d6bc1ea0508cac0323817768a5dbe/bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/IndegoController.java#L159-L173

Reply:
Yep, thats it! I like your solution better i think, i might go back at some point and replace mine when i have time to test it out.